tests/foreman/cli/test_repository.py

```diff
@@ -1077,6 +1098,26 @@ def test_positive_update_gpg_key(self):
         result = Repository.info({'id': new_repo['id']})
         self.assertEqual(result['gpg-key']['id'], gpg_key_new['id'])
 
+    @tier1
+    def test_positive_update_mirror_on_sync(self):
+        """Update the mirror on sync rule for repository
+
+        :id: 9bab2537-3223-40d7-bc4c-a51b09d2e812
+
+        :expectedresults: Repository is updated
+
+        :CaseImportance: Critical
+        """
+        new_repo = self._make_repository({
+            u'mirror-on-sync': 'no',
+        })
+        Repository.update({
+            u'id': new_repo['id'],
+            u'mirror-on-sync': 'yes',
+        })
+        result = Repository.info({'id': new_repo['id']})
+        self.assertEqual(result['mirror-on-sync'], 'yes')
```

Comment:
nitpick: please use subtest in a similar to `create` test way and ensure mirror-on-sync can be switched back to 'no'.
We had issues with this parameter not being able to switch from one state to another correctly; it's not the case here as you aren't really verifying how it's working, but imo it's not completely useless and worth covering.